wallet/src/main.rs

```diff
@@ -89,18 +121,83 @@ async fn main() -> anyhow::Result<()> {
+        println!("Keystore wsa empty. Inserting example key for THIS SESSION ONLY");
+
+        // This only inserts it into memory. That should be fine for the example key since it can always be
+        // re-inserted on each new run. But for user-provided keys, we want them to be persisted.
+        // Hopefully insert_unknown will make that happen.
+        keystore
+            .sr25519_generate_new(KEY_TYPE, Some(SHAWN_PHRASE))
+            .map_err(|e| anyhow!("{:?}", e))?;
+    }
+
     // Setup jsonrpsee and endpoint-related information.
     // https://github.com/paritytech/jsonrpsee/blob/master/examples/examples/http.rs
     let client = HttpClientBuilder::default().build(cli.endpoint)?;
 
     // Dispatch to proper subcommand
     match cli.command {
         Command::AmoebaDemo => amoeba::amoeba_demo(&client).await,
-        Command::VerifyCoin { ref_string } => {
-            let output_ref = OutputRef::decode(&mut &hex::decode(ref_string)?[..])?;
+        Command::VerifyCoin { output_ref } => {
             money::print_coin_from_storage(&output_ref, &client).await
         }
-        Command::SpendCoins(args) => money::spend_coins(&client, args).await,
+        Command::SpendCoins(args) => money::spend_coins(&client, &keystore, args).await,
+        Command::InsertKey { seed } => {
+            // We need to provide a public key to the keystore manually, so let's calculate it.
+            let public_key = Pair::from_phrase(&seed, None)?.0.public();
+            println!("The generated public key is {:?}", public_key);
+            keystore
+                .insert_unknown(KEY_TYPE, &seed, public_key.as_ref())
+                .map_err(|e| anyhow!("{:?}", e))?;
+            Ok(())
+        }
+        Command::ShowKeys => {
+            keystore
+                .keys(KEY_TYPE)?
+                .into_iter()
+                .filter_map(|CryptoTypePublicPair(t, public)| {
+                    // Since we insert with `insert_unknown`, each key is inserted three times.
+                    // Here we filter out just the sr25519 variant so we don't print duplicates.
+                    if t == CryptoTypeId(*b"sr25") {
+                        Some(public)
+                    } else {
+                        None
+                    }
+                })
+                .for_each(|pubkey| {
+                    println!("key: 0x{}", hex::encode(pubkey));
+                });
+
+            Ok(())
+        }
+        Command::RemoveKey { pub_key } => {
+            // The keystore doesn't provide an API for removing keys, so we
+            // remove them from the filesystem directly
+            let filename = format!("{}{}", hex::encode(KEY_TYPE.0), hex::encode(pub_key.0));
+            let path = keystore_path.join(filename);
+
+            println!("CAUTION!!! About permanently remove {pub_key}. This action CANNOT BE REVERSED. Type \"proceed\" to confirm deletion.");
+
+            let mut confirmation = String::new();
```

Comment:
Ok good I was going to suggest this atleast a confirmation is one safeguard that can help with unwanted things such as deleting keys by accident